src/constraints/emissions_constraints.jl

```diff
@@ -0,0 +1,156 @@
+# *********************************************************************************
+# REopt, Copyright (c) 2019-2020, Alliance for Sustainable Energy, LLC.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without modification,
+# are permitted provided that the following conditions are met:
+#
+# Redistributions of source code must retain the above copyright notice, this list
+# of conditions and the following disclaimer.
+#
+# Redistributions in binary form must reproduce the above copyright notice, this
+# list of conditions and the following disclaimer in the documentation and/or other
+# materials provided with the distribution.
+#
+# Neither the name of the copyright holder nor the names of its contributors may be
+# used to endorse or promote products derived from this software without specific
+# prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+# OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+# OF THE POSSIBILITY OF SUCH DAMAGE.
+# *********************************************************************************
+
+function add_emissions_constraints(m,p)
+	if !isnothing(p.s.site.CO2_emissions_reduction_min_pct)
+		@constraint(m, MinEmissionsReductionCon, m[:Lifecycle_Emissions_Lbs_CO2] <= (1-p.s.site.CO2_emissions_reduction_min_pct)*m[:Lifecycle_Emissions_Lbs_CO2_BAU])
+	end
+	if !isnothing(p.s.site.CO2_emissions_reduction_max_pct)
+		@constraint(m, MaxEmissionsReductionCon, m[:Lifecycle_Emissions_Lbs_CO2] >= (1-p.s.site.CO2_emissions_reduction_max_pct)*m[:Lifecycle_Emissions_Lbs_CO2_BAU])
+	end
+end
+
+### Year 1 Emissions Calculations
+function add_yr1_emissions_calcs(m,p)
+	# Components:
+	m[:yr1_emissions_onsite_fuel_lbs_CO2], m[:yr1_emissions_onsite_fuel_lbs_NOx], 
```

Comment:
@hdunham @NLaws do you think it would be worth indexing all of these emissions model vars and associated parameters on pollutant, rather than have each one individually named? Pros: significantly decrease the length of code. Cons: will take some time. 
e.g.
- `p.tech_emissions_factors_CO2[t]` --> `p.tech_emissions_factors[p][t]` where p in P, set [CO2, NOx, SO2, PM2.5] 
- `m[:yr1_emissions_onsite_fuel_lbs_CO2]` --> `m[:yr1_emissions_onsite_fuel_lbs][p]`

Reply:
I do like the sound of that. Not the highest priority though, so will probably merge without that and change later when funds aren't so tight. I don't think we would change that in the input dict, just how they are saved in the structs.